### PR TITLE
Alerting: Fix test fails in some environments

### PR DESCRIPTION
**Pull request by grobinson-grafana**
**Which issue(s) does this PR fix?**:

A quick fix for failing tests in some environments.

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/ngalert/models/time.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	// timeNow is an equivalent time.Now() that can be replaced in tests
+	timeNow = time.Now
+)
+
+func newFixedTimeNow() func() time.Time {
+	return func() time.Time {
+		tm, _ := time.Parse(time.RFC3339, "2022-01-01T00:00:00")
+		return tm
+	}
+}
```
Comment (yuri-tceretian) [inline pkg/services/ngalert/models/time.go:15]: I do not think that this needs to be in production code. This is 100% testing

File: pkg/services/ngalert/models/image_test.go [hunk 2]
```diff
@@ -20,12 +20,18 @@ func TestImage_ExtendDuration(t *testing.T) {
...
+	t.Cleanup(func() {
+		timeNow = oldTimeNow
+	})
+
 	var i Image
-	i.ExpiresAt = time.Now().Add(time.Minute)
+	i.ExpiresAt = timeNow().Add(time.Minute)
 	assert.False(t, i.HasExpired())
-	i.ExpiresAt = time.Now()
+	i.ExpiresAt = timeNow().Add(-time.Millisecond)
```
Comment (yuri-tceretian) [inline pkg/services/ngalert/models/image_test.go:32]: This test should check whether `ExpiresAt = timeNow` is considered expired. Your change is basically the same as few lines below

**Comment by JohnnyQQQQ** [comment]
We already have `github.com/benbjohnson/clock` in Grafana in use, I think this might be the more elegant way of solving this than introducing some global state, wdyt?

**Comment by grobinson-grafana** [comment]
> We already have `github.com/benbjohnson/clock` in Grafana in use, I think this might be the more elegant way of solving this than introducing some global state, wdyt?

You still need a shared `time.Time` variable right?

**Comment by grobinson-grafana** [comment]
> We already have `github.com/benbjohnson/clock` in Grafana in use, I think this might be the more elegant way of solving this than introducing some global state, wdyt?

We would still need a shared `clock.Clock` variable so it can be modified in the tests?

**Review by yuri-tceretian — APPROVED** [review]
LGTM

**Pull request merged**